cmd/capabilities/worker/worker.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+        "log"
+        "fmt"
+        "net/http"
+        "os"
+        "io/ioutil"
+	    "os/exec"
+	    "strings"
+        "github.com/apigee/registry/cmd/capabilities/utils"
+        "encoding/json"
+        "cloud.google.com/go/compute/metadata"
+)
+
+func main() {
+    log.Print("starting server...")
+    http.HandleFunc("/", requestHandler)
+
+    // Determine port for HTTP service.
+    port := os.Getenv("PORT")
+    if port == "" {
+            port = "8080"
+            log.Printf("defaulting to port %s", port)
+    }
+
+    // Start HTTP server.
+    log.Printf("listening on port %s", port)
+    if err := http.ListenAndServe(":"+port, nil); err != nil {
+            log.Fatal(err)
+    }
+}
+
+func getAuthToken() (string, error) {
+    serviceURL := "http://" + os.Getenv("APG_REGISTRY_ADDRESS")
+    tokenURL := fmt.Sprintf("/instance/service-accounts/default/identity?audience=%s", serviceURL)
+    idToken, err := metadata.Get(tokenURL)
+    if err != nil {
+            log.Printf("metadata.Get: failed to query id_token: %+v", err)
+            return "", err
+    }
+
+    return idToken, nil
+}
+
+func requestHandler(w http.ResponseWriter, r *http.Request) {
+    body, err := ioutil.ReadAll(r.Body)
+    if err != nil{
+        log.Printf("ioutil.ReadAll: %v", err)
+        http.Error(w, "Bad Request", http.StatusBadRequest)
+        return
+    }
+
+    req := utils.WorkerRequest{}
+    err = json.Unmarshal(body, &req)
+    if err != nil {
+       log.Printf("json.Unmarshal: %v", err)
+       http.Error(w, "Bad Request", http.StatusBadRequest)
```

Comment:
Similar to above, can we indicate that the request JSON was invalid or provide any other information in the HTTP response?

Reply:
Done.